headlamp-k8s/plugins
# backstage: Init backstage plugin

Adds backstage plugin for headlamp, In the details view of a resource if the resource has labels related to backstage an Icon Button is added to open the corresponding backstage service. The plugin also provides a way to configure the backstage url for each cluster. If headlamp with this plugin is used as an iframe it sends a message to the parent window with the path to redirect to for the parent to handle and redirect. 

Steps to test (Without running backstage):
1. Run the following command to create resources with backstage related labels
> kubectl apply -f https://raw.githubusercontent.com/backstage/backstage/refs/heads/master/plugins/kubernetes-backend/examples/dice-roller/dice-roller-manifests.yaml

2. Run the backend,frontend and the backstage plugin.
3.  Navigate to the plugin settings and configure URL and configure the value for the cluster to say `http://google.com`
![image](https://github.com/user-attachments/assets/6214bb32-f40b-42c9-8a13-d34d05eb234c)
4. Navigate to the detail view of any of the dice-roller resource, Click on the backstage button.
![image](https://github.com/user-attachments/assets/43b812d2-b89a-4e70-bec9-df5b7902bc71)
5. Click on the backstage button a new window should be opened with the following url
`http://google.com/catalog/default/component/dice-roller`
